owls.py

```diff
@@ -85,9 +67,25 @@ def fill_details(key, item, task, connection):
         #For non-Thing objects, ask for further detail
         if key != 'Thing':
             #Ask for label
-            obj_name = raw_input(key + " name/title: ")
+            if key == 'Author':
```

Comment:
Maybe do separate handler functions for each thing that is selectable? Depending on how you write the code and how the modules for different things are constructed it could automatically pull the `cli_run_query` function and then each module would define its own cli interaction